app/com/m3/octoparts/repository/ConfigsRepository.scala

```diff
@@ -25,14 +25,34 @@ trait ConfigsRepository {
   def findParamById(id: Long): Future[Option[PartParam]]
 
   /**
-   * Find a single ThreadPoolConfig by id
+   * Find a single ThreadPoolConfig by id.
+   *
+   * All retrieved items are populated with their [[HystrixConfig]]
    */
-  def findThreadPoolConfigById(id: Long): Future[Option[ThreadPoolConfig]]
+  def findThreadPoolConfigById(id: Long): Future[Option[ThreadPoolConfig]] = findAllConfigs().map {
+    configs =>
+      {
+        val tpcMap = configs.collect { case part if part.hystrixConfigItem.threadPoolConfigItem.id.get == id => part.hystrixConfigItem.threadPoolConfigItem -> part.hystrixConfigItem } groupBy (_._1) mapValues (_.map(_._2))
+        tpcMap.headOption.map {
+          case (tpc, hystrixConfigs) => tpc.copy(hystrixConfigs = hystrixConfigs)
+        }
+      }
+  }(ExecutionContext.global)
 
   /**
    * Return all the thread pool configs
+   *
+   * All retrieved items are populated with their [[HystrixConfig]]
    */
-  def findAllThreadPoolConfigs(): Future[Seq[ThreadPoolConfig]]
+  def findAllThreadPoolConfigs(): Future[Seq[ThreadPoolConfig]] = findAllConfigs().map {
```

Comment:
Leaving a comment for this method, so it doesn't overlap w/ the above (but it applies to both):

Please add tests for these 2 methods to make sure they actually work as advertised (populated w/ their (and only their) HystrixConfigs).